src/app/shared/services/remote-asset/remote-asset.service.ts

```diff
@@ -0,0 +1,109 @@
+import { Injectable } from "@angular/core";
+import { SyncServiceBase } from "../syncService.base";
+import { AppConfigService } from "../app-config/app-config.service";
+import { IAppConfig } from "../../model";
+import { createClient, SupabaseClient } from "@supabase/supabase-js";
+import { FileManagerService } from "../file-manager/file-manager.service";
+import { TemplateActionRegistry } from "../../components/template/services/instance/template-action.registry";
+
+@Injectable({
+  providedIn: "root",
+})
+export class RemoteAssetService extends SyncServiceBase {
```

Comment:
nit(non-blocking)
I expect lots of work will be needed on this in the future so no need to focus on this for now. But as a rough outline makes sense to me, good to see remote asset management separate to file management services